Kernel_23/include/CGAL/Exact_predicates_inexact_constructions_kernel.h

```diff
@@ -19,28 +19,80 @@
 
 #include <CGAL/Simple_cartesian.h>
 #include <CGAL/Filtered_kernel.h>
+#include <CGAL/Converting_construction.h>
 #include <CGAL/Triangulation_structural_filtering_traits.h>
 
 namespace CGAL {
 
-// The following is equivalent to Filtered_kernel< Simple_cartesian<double> >,
-// but it's shorter in terms of template name length (for error messages, mangling...).
-
-class Epick
-  : public Filtered_kernel_adaptor<
-               Type_equality_wrapper< Simple_cartesian<double>::Base<Epick>::Type, Epick >,
+constexpr bool epick_use_static_filter =
 #ifdef CGAL_NO_STATIC_FILTERS
-               false >
+    false;
 #else
-               true >
+    true;
 #endif
+
+#ifndef CGAL_EPICK_SINGLE_PRECISION
+// Here Epick is a class, and Double_precision_epick an alias to it.
+class Epick;
+class Single_precision_epick;
+using Double_precision_epick = Epick;
+#else // CGAL_EPICK_SINGLE_PRECISION
+// Here, Epick is an alias to Single_precision_epick.
+class Single_precision_epick;
+class Double_precision_epick;
+using Epick = Single_precision_epick;
+#endif // CGAL_EPICK_SINGLE_PRECISION
```

Comment:
We discussed that IRL as well, and we still do not completely agree.

My goal is to make sure `CGAL::Epick` does not change its behavior and semantic, unless `CGAL_EPICK_SINGLE_PRECISION`. Making `CGAL::Epick` an alias for `Double_precision_epick` would break forward declarations of it.

But, if `CGAL_EPICK_SINGLE_PRECISION` is defined, then I want to make sure that cannot be undetected. And then `CGAL::Epick` becomes an alias to `Single_precision_epick`.

Reply:
How about something like:
```
class Single_precision_epick : ... etc.
class Double_precision_epick : ... etc.
#ifdef CGAL_EPICK_SINGLE_PRECISION
class Epick : public Single_precision_epick { };
#else
class Epick : public Double_precision_epick { };
#endif

template <>
struct Triangulation_structural_filtering_traits<Epick> {
  using Use_structural_filtering_tag = Tag_true;
};
```
should be fine?

Maybe I misunderstood what you meant with "cannot be undetected".

Reply:
With
```c++
class Epick : public Double_precision_epick { };
```
then `Epick` looses the "type equality property": `Epick::Point_2` and `Point_2<Epick>` would be different.

Reply:
> Maybe I misunderstood what you meant with "cannot be undetected".

With the current implementation, if `CGAL_EPICK_SINGLE_PRECISION` is defined, then the debug symbols (and call stacks in compiler warnings) mention `CGAL::Simple_precision_epick` instead of `CGAL::Epick`. That helps debugging, in my opinion. It might help a user remembering that he/she has defined `CGAL_EPICK_SINGLE_PRECISION`, and it might help us if a user interacts with us about a bug.

Reply:
> With
> 
> ```c++
> class Epick : public Double_precision_epick { };
> ```
> 
> then `Epick` looses the "type equality property": `Epick::Point_2` and `Point_2<Epick>` would be different.

Right. However, one can still define a class in both cases.

I am not opposed to the current setup; I only think that the asymmetry is a cost to pay when someone has to understand that code later in the future.

Reply:
Done